ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -285,99 +296,122 @@ storeLedgerStateAt ::
      forall blk .
      ( LgrDbSerialiseConstraints blk
      , HasAnalysis blk
-     , LedgerSupportsProtocol blk
      )
-  => SlotNo -> Analysis blk
-storeLedgerStateAt slotNo (AnalysisEnv { db, registry, initLedger, cfg, limit, ledgerDbFS, tracer }) =
-    void $ processAllUntil db registry GetBlock initLedger limit initLedger process
+  => SlotNo
+  -> Analysis blk
+storeLedgerStateAt slotNo AnalysisEnv { db, backing, initLedger, cfg, ledgerDbFS, tracer } = do
+
+    tp <- atomically $ either ImmutableDB.getTipPoint ChainDB.getTipPoint db
+    when (pointSlot tp < At slotNo) $ warning (pointSlot tp)
+
+    initLedger' <- initialLedger ledgerDbFS cfg initLedger
+
+    tracer' <- decorateReplayTracerWithStart (castPoint . getTip $ initLedger') . mkTracer <$> newIORef (0 :: Int)
+
+    (initDb, backingStore) <- ledgerDbAndBackingStore backing ledgerDbFS $ case initLedger of
+                                                                             Left s -> Left (initLedger', s)
+                                                                             Right v -> Right v
+
+    eDB <- runExceptT
+         $ replayStartingWith
+             tracer'
+             configLedgerDb
+             backingStore
+             (mkStream (return . (>= slotNo) . blockSlot) db)
+             initDb
+
+    case eDB of
+      Left e -> error $ show e
+      Right (ldb, _) -> do
+        OnDisk.flush backingStore $ ledgerDbChangelog ldb
+
+        writeSnapshot
```

Comment:
I also wonder whether this function should perform the flush. I think we discussed this in the past but I don't recall the outcome.